### DISCUSSION TITLE
Using RAC's TextField + React Hook Form

### DISCUSSION THREAD

**Discussion by thevipinmishra** [adobe/react-spectrum#5089] [Q&A]
I was giving the [TextField](https://react-spectrum.adobe.com/react-aria/TextField.html) component a try, along with [React Hook Form](https://react-hook-form.com/).

Now, adding the props returned by `register` function into the `TextField` component is not working as expected, I know this can easily be solved by spreading the props on the `Input` element that we nest under the TextField anyway, but was curious to know about the API and design decisions.

```js
{name: 'companyName', onChange: ƒ, onBlur: ƒ, ref: ƒ}
```

Passing props to TextField shouldn't just work?

**Comment by eashish93** [comment]
Use `Controller` of react-hook-form like this: 
```js

<Controller
  name="url"
  control={control}
  render={({ field }) => (
    <TextField
      label="URL"
      {...field}
    />
  )}
/>

```
Using `register` with `<TextField>` is not working because `onChange` prop is different for both while using with `Controller`, it does match.

**Reply by hosmelq** [reply]
I'm doing the same.

**Comment by thevipinmishra** [comment]
> Use `Controller` of react-hook-form like this:
> 
> ```js
> <Controller
>   name="url"
>   control={control}
>   render={({ field }) => (
>     <TextField
>       label="URL"
>       {...field}
>     />
>   )}
> />
> ```
> 
> Using `register` with `<TextField>` is not working because `onChange` prop is different for both while using with `Controller`, it does match.

Any documentation?

I suppose, it was only for components which are meant to be Controlled only, like MUI.

**Reply by snowystinger** [reply]
you mean this https://react-hook-form.com/get-started/#IntegratingwithUIlibraries ?

**Reply by eashish93** [reply]
@thevipinmishra , also see this link to check controller `onChange` method: https://react-hook-form.com/docs/usecontroller/controller
